crates/agentgateway/src/types/agent.rs

```diff
@@ -1030,6 +1030,60 @@ pub struct TargetedPolicy {
 	pub policy: PolicyType,
 }
 
+/// Configuration for dynamic tracing policy
+#[apply(schema_ser!)]
+pub struct TracingConfig {
+	pub service_name: String,
+	pub provider_backend: SimpleBackendReference,
+	pub attributes: Vec<TracingAttribute>,
+}
+
+/// A single tracing attribute with a CEL expression
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "schema", derive(JsonSchema))]
+pub struct TracingAttribute {
+	pub name: String,
+	#[cfg_attr(feature = "schema", schemars(skip))]
+	pub value: Arc<cel::Expression>,
+}
+
+impl serde::Serialize for TracingAttribute {
```

Comment:
why do we need a cusotm Serialize? including the value expression seems useful and simpler